pkg/local_object_storage/metabase/exists.go

```diff
@@ -86,14 +93,50 @@ func (db *DB) exists(tx *bbolt.Tx, addr oid.Address) (exists bool, err error) {
 	return firstIrregularObjectType(tx, cnr, objKey) != objectSDK.TypeRegular, nil
 }
 
-// inGraveyard returns:
-//  * 0 if object is not marked for deletion;
+var errStopIteration = errors.New("stop")
+
+// objectStatus returns:
+//  * 0 if object is available;
 //  * 1 if object with GC mark;
-//  * 2 if object is covered with tombstone.
-func inGraveyard(tx *bbolt.Tx, addr oid.Address) uint8 {
+//  * 2 if object is covered with tombstone;
+//  * 3 if object is expired.
+func objectStatus(tx *bbolt.Tx, addr oid.Address, currEpoch uint64) uint8 {
+	// we check only if the object is expired in the current
+	// epoch since it is considered the only corner case: the
+	// GC is expected to collect all the objects that have
+	// expired previously for less than the one epoch duration
+
+	rawOID := []byte(addr.Object().EncodeToString())
+	var expired bool
+
+	// bucket with objects that have expiration attr
+	expirationBucket := tx.Bucket(attributeBucketName(addr.Container(), objectV2.SysAttributeExpEpoch))
+	if expirationBucket != nil {
+		// bucket that contains objects that expire in the current epoch
+		currEpochBkt := expirationBucket.Bucket([]byte(strconv.FormatUint(currEpoch, 10)))
+		if currEpochBkt != nil {
+			// we can do nothing with errors from db
+			// => consider object as not expired
+			_ = currEpochBkt.ForEach(func(k, _ []byte) error {
```

Comment:
Why simple `Get` doesn't work?

Reply:
Also, storage errors _should_ be returned to the caller.

Reply:
> Why simple Get doesn't work?

no idea, fixed

> Also, storage errors should be returned to the caller.

yes, but that func returns just `uint8`. fixed to `Get` call anyway